### PR TITLE
add Dark Theme support

### LINKED ISSUE #17: Feature Request: Dark Mode
A simple dark/AMOLED black mode that gets triggered via the System theme would be appreciated.

### PR DESCRIPTION
**Pull request by S-H-Y-A**
Fixes #17
Please take a look at the pictures.

Screenshot_20210721-195122783
Screenshot_20210721-195142716

### CODE REVIEW

**Comment by sakusaku3939** [comment]
Thanks for the good work!
Basically it seems to be working fine, but the app is crashing when certain actions are performed, so please fix this.

File: app/src/main/java/com/example/deeplviewer/MainActivity.kt [hunk 1]
```diff
@@ -85,6 +86,25 @@ class MainActivity : AppCompatActivity() {
...
+        val webView: WebView = findViewById(R.id.webview)
+        val defParamValue = "#en/en/"
+        val urlParam = getSharedPreferences("config", Context.MODE_PRIVATE).getString(
+            "urlParam",
+            defParamValue
+        ) ?: defParamValue
+        val startUrl = "https://www.deepl.com/translator$urlParam"
+        val inputText =
+            Uri.decode((webView.url)
+                ?.substring(startUrl.length))
```
Comment (sakusaku3939) [inline app/src/main/java/com/example/deeplviewer/MainActivity.kt:100]: Throws `java.lang.StringIndexOutOfBoundsException` if `onSaveInstanceState` is called after all characters have been deleted

Examples
1) Input a character
2) Delete all characters
3) Switching themes

My guess is that the "#ja/en/" parameter is lost when the input field is empty, so you have specified an index out of bounds.

**Review by sakusaku3939 — CHANGES_REQUESTED** [review]
Thanks for the good work!
Basically it seems to be working fine, but the app is crashing when certain actions are performed, so please fix this.

File: app/src/main/java/com/example/deeplviewer/MainActivity.kt (hunk 1 above)
Comment (S-H-Y-A) [reply app/src/main/java/com/example/deeplviewer/MainActivity.kt]: I changed to compare webView url length and default url length to solve this problem.

**Comment by sakusaku3939** [comment]
Seems to be no problem!
Lastly, I'll make some corrections on my end so that the splash screen logo and the connection failure screen logo will also change to match the dark mode.

Screenshot:
Screenshot_20210724-012115

**Comment by adarsh1001** [comment]
Thanks for this! @S-H-Y-A @sakusaku3939  one small suggestion would be to color the top status bar as well so that it blends with the dark mode (like other apps do).

Google Translate Dark Mode:


Current DeepL Dark Mode:

**Pull request merged**